flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarHandlerParameterTest.java

```diff
@@ -0,0 +1,300 @@
+	}
+
+	private void testConfigurationViaQueryParameters(ProgramArgsParType programArgsParType) throws Exception {
+		// configure submission via query parameters
+		handleRequest(createRequest(
+			getDefaultJarRequestBody(),
+			getJarMessageParameters(programArgsParType),
+			getUnresolvedJarMessageParameters(),
+			jarWithoutManifest));
+		validateGraph();
+	}
+
+	@Test
+	public void testConfigurationViaJsonRequestWithProgArgsAsString() throws Exception {
+		testConfigurationViaJsonRequest(ProgramArgsParType.String);
+	}
+
+	@Test
+	public void testConfigurationViaJsonRequestWithProgArgsAsList() throws Exception {
+		testConfigurationViaJsonRequest(ProgramArgsParType.List);
+	}
+
+	@Test
+	public void testConfigurationViaJsonRequestFailWithProgArgsAsStringAndList() throws Exception {
+		try {
+			testConfigurationViaJsonRequest(ProgramArgsParType.Both);
+			fail("RestHandlerException is excepted");
+		} catch (RestHandlerException e) {
+			assertEquals(HttpResponseStatus.BAD_REQUEST, e.getHttpResponseStatus());
+		}
+	}
+
+	private void testConfigurationViaJsonRequest(ProgramArgsParType programArgsParType) throws Exception {
+		handleRequest(createRequest(
+			getJarRequestBody(programArgsParType),
+			getUnresolvedJarMessageParameters(),
+			getUnresolvedJarMessageParameters(),
+			jarWithoutManifest
+		));
+		validateGraph();
+	}
+
+	@Test
+	public void testParameterPrioritizationWithProgArgsAsString() throws Exception {
+		testParameterPrioritization(ProgramArgsParType.String);
+	}
+
+	@Test
+	public void testParameterPrioritizationWithProgArgsAsList() throws Exception {
+		testParameterPrioritization(ProgramArgsParType.List);
+	}
+
+	@Test
+	public void testFailIfProgArgsAreAsStringAndAsList() throws Exception {
+		try {
+			testParameterPrioritization(ProgramArgsParType.Both);
+			fail("RestHandlerException is excepted");
+		} catch (RestHandlerException e) {
+			assertEquals(HttpResponseStatus.BAD_REQUEST, e.getHttpResponseStatus());
+		}
+	}
+
+	private void testParameterPrioritization(ProgramArgsParType programArgsParType) throws Exception {
+		// configure submission via query parameters and JSON request, JSON should be prioritized
+		handleRequest(createRequest(
+			getJarRequestBody(programArgsParType),
+			getWrongJarMessageParameters(programArgsParType),
+			getUnresolvedJarMessageParameters(),
+			jarWithoutManifest));
+		validateGraph();
+	}
+
+	static String getProgramArgsString(ProgramArgsParType programArgsParType) {
+		return programArgsParType == ProgramArgsParType.String || programArgsParType == ProgramArgsParType.Both
+			? String.join(" ", PROG_ARGS) : null;
+	}
+
+	static List<String> getProgramArgsList(ProgramArgsParType programArgsParType) {
+		return programArgsParType == ProgramArgsParType.List || programArgsParType == ProgramArgsParType.Both
+			? Arrays.asList(PROG_ARGS) : null;
+	}
+
+	private static <
+		REQB extends JarRequestBody,
+		M extends JarMessageParameters>
+	HandlerRequest<REQB, M> createRequest(
```

Comment:
the indentation doesn't look correct, it should not be in the same column as `private static <`